lib/molinillo/resolution.rb

```diff
@@ -364,6 +364,16 @@ def fixup_swapped_children(vertex)
             debug(depth) { "Removing orphaned spec #{succ.name} after swapping #{name}" }
             activated.detach_vertex_named(succ.name)
             requirements.delete_if { |r| name_for(r) == succ.name }
+            # TODO:
+            # 1. I'm not sure if it's correct to be deleting these
+            # requirements. Does not supporting circular deps ensure that the
+            # requirement must have come from `succ`? What about diamond-shaped
+            # dependency graphs?
+            # 2. If it's okay to delete the requirements, this should be
+            # recursive.
+            succ.successors.each do |succ_dep|
```

Comment:
are you sure this shouldn't be `predecessors`? also, this can be folded into the prior `delete_if` loop by using an `include?` check